ui-v2/src/components/concurrency/global-concurrency-view/index.tsx

```diff
@@ -1,28 +1,58 @@
 import { useListGlobalConcurrencyLimits } from "@/hooks/global-concurrency-limits";
-import { useState } from "react";
-
+import { useMemo, useState } from "react";
+import { CreateOrEditLimitDialog } from "./create-or-edit-limit-dialog";
+import { GlobalConcurrencyLimitEmptyState } from "./global-concurrency-limit-empty-state";
 import { GlobalConcurrencyLimitsHeader } from "./global-concurrency-limits-header";
 
+type AddOrEditDialogState = {
+	open: boolean;
+	limitIdToEdit?: string;
+};
+
 export const GlobalConcurrencyView = () => {
-	const [showAddDialog, setShowAddDialog] = useState(false);
+	const [openAddOrEditDialog, setOpenAddOrEditDialog] =
+		useState<AddOrEditDialogState>({
+			open: false,
+		});
```

Comment:
I had some trouble wrapping my head around this on the first couple of reads. This suggestion I think it a little more clear, but I'm open to alternatives.
```suggestion
	const [addOrEditDialogState, addOrEditDialogState] =
		useState<AddOrEditDialogState>({
			open: false,
			limitIdToEdit: undefined
		});
```

Reply:
Based on your comment below here: https://github.com/PrefectHQ/prefect/pull/16248/files#r1873864935, this makes sense.

I was trying to represent the dialog state to:
1. is it open?
2. Should it open with a limit to be edited?

Reply:
Updated in latest push.

Simplified it to it just keeps track of it being open.
I removed the selected row logic and will handle it when doing the table